### DISCUSSION TITLE
Regarding Inhibitory Neurons and Excitatory Neuron Trainings Under Dale's Rule

### DISCUSSION THREAD

**Discussion by Arueruma1999** [BindsNET/bindsnet#680] [Q&A] [answered]
Hi!

I'm recently trying to create a network with inhibitory neurons and excitatory neurons training under Dale's rule, where weights connecting neurons would not change their sign during training. I'm wondering if there is a way I could easily do this with some parameters passed to node or connection class.

Thanks alot in advance!

**Comment by Hananel-Hazan** [comment, marked as answer]
I've incorporated Dale's rule into my branch, which is a valuable addition. To utilize it, simply include the Dales_rule parameter in the connection object. This parameter accepts a Boolean matrix with the same dimensions as weights, where True represents a positive connection and False indicates a negative connection. It's important to note that implementing this rule may significantly slow down the runtime. I may work on optimizing it for better performance in the future. Feel free to try it out and let me know your thoughts.

**Reply by Arueruma1999** [reply]
Sure, I'll try it out recently. Thank you for your help!
